usermanual_en.html

octaeder opened the pull request:
fully revised section 3.2 The log files

sunderme commented on utilities/manual/usermanual_en.html:
```diff
@@ -548,14 +548,13 @@ <h1 id="SECTION2">3. Compiling a document</h1>
 <p>The easiest way to compile a document is to use the "Compile" command or the "Build&amp;View" command ("Compile" button - shortcut : F6). You can select the default command via the "Configure TeXstudio" dialog.<br>(You can also launch each command one by one in the "Tools" menu).<br> Note : the "Clean" command in the "Tools menu" allows you to erase the files (dvi, toc, aux...) generated by a  ...
 <p><IMG src="compile_toolbar.png" border="1" alt="compile_toolbar">
-<p><b>Warning: all your files must have an extension and you can't compile an "untitled" file or a file with a space in its name</b>.</p>
+<p><b>Warning: all your files must have an extension and you can't compile a file with a space in its name</b>.</p>
```
that is not true any more (for some time)
filename may contain spaces

octaeder replied:
so we stay with a warning that all files must have an extension? or remove completely?

sunderme commented on utilities/manual/usermanual_en.html:
```diff
@@ -548,14 +548,13 @@ <h1 id="SECTION2">3. Compiling a document</h1>
-A tool tip will show more details of the mistake if you jump to a line (it is also shown if you move the mouse over the mark left from the line numbers).</p>
+<p>The Log tab helps with error handling. It provides the tabs Log File and Issues as well as some controls (see figure). The Issues tab displays a table of messages taken from the log file displayed below the table.</p>
+<p><IMG src="errorlog.png" border="1" alt="Display of Error Log"></p>
+<p>The first three controls can be used to choose whether errors (red), warnings (yellow) or bad boxes (blue) are shown, another whether to show error markers in the editor, and the last two to toggle back and forth between errors to be jumped. The shortcuts for this are Ctrl+Shift+Up/Down, accordingly Ctrl+Alt+Up/Down and Alt+Shift+Up/Down are used for warnings and bad boxes respectively. You ju ...
```
I would not number the buttons, especially as all are provided with tooltips.
Use the tooltips name (ideally with icon images, but that quickly get tedious...)

octaeder replied:
The usage of the dialog is quite clear and the description just gives the user some basic understanding and orientation. And there is other work waiting. I'll add the icon tool tips...

octaeder commented:
can this be merged or is something more to do?

sunderme commented:
so there are still a number of open issues for this.
How do we proceed ?

octaeder commented:
Round up:

- "file extensions" & "filenames with spaces": unanswered question. felt have to decide on my on. removed sentence.
- "use tooltips instead of numbers": Changed from numbers to tooltip names (highlighted in italic)". didn't use icons inline, cause you named it a tedious work (I agree) and style nowhere else used in this manual before. Text describes clearly where to find icons in app/image, so reader can follow easily description of buttons (same order as in image left to right).
- "Ctr+Alt+Up/Down": opened review, no answer. still in text. don't no why described but not implemented. suggest introduce as default shortcut (currently not my part).
- Meanwhile added description of dialog automatisms and of 3 options from config dialog.
- Added today pending change: Ctrl+Up/Down jumps between all markers (incl. bookmarks).

From my side ready for production, when shortcut will come. Otherwise will remove description of specific shortcut or add suggestion which missing shortcut could be set up on his/her own.

sunderme commented:
i have started a review and commented on some issues. I assume that you get the remarks as mail, but in any case, scroll up on this page.
Pending reviews state what I  consider missing.

sunderme commented on utilities/manual/usermanual_en.html:
```diff
@@ -548,14 +548,13 @@ <h1 id="SECTION2">3. Compiling a document</h1>
+<p><IMG src="errorlog.png" border="1" alt="Display of Error Log"></p>
+<p>With the buttons <i>Show Error</i>, <i>Show Warning</i>, <i>Show BadBox</i> (see tooltips) you can choose whether error messages (red), warning messages (yellow) or messages for bad boxes (blue) will be shown. With the button <i>Show Log Markers</i> you can display log marker icons left to the lines in the editor. The tooltips for the log markers show details about the errors if required.</p>
+<p>When there are compile error messages then the log markers will be activated. In addition if option <i>Show log in case of compile error</i> (s. Build settings) is checked then the Log tab will be opened automatically. If option <i>Go to error when displaying log</i> is checked (s. Adv. Editor settings) then the editor's cursor will be placed in the first line which has an error marker.</p>
+<p>You may use the buttons <i>Previous Error</i> and <i>Next Error</i> to toggle back and forth between errors to be jumped. The shortcuts for this are Ctrl+Shift+Up/Down, accordingly Ctrl+Alt+Up/Down and Alt+Shift+Up/Down are used for warnings and bad boxes respectively. You jump unspecifically between markers (including bookmarks) with Ctrl+Up/Down.</p>
```
ctrl+up/down is go to prev/next mark. It jumps to the next annotated icon (left of the text) That can be bookmarks but also marked errors or warnings.

sunderme commented:
maybe I missed to submit the review, so sorry about that

octaeder commented:
yes, you didn't send it. And after thinking for a long time here if I should send my answer I throw it away now. And I'm thinking about doing the same with my changes here. Can't see any way to predict what you want.

sunderme commented:
Sorry, that I accidentally did not respond in time.
However the remarks above are true for any good documentation.

octaeder commented:
@sunderme Please check all my new annotations above.

octaeder commented:
To avoid changing back and forth I added my suggestions above. If you think that the last changes described above have a high propability that you will agree to them when they are integrated, then let me know and I will do so.

sunderme commented:
I don't see any answers to the my review.
![grafik](https://user-images.githubusercontent.com/14033169/167161857-5c575153-66b5-4650-bbfa-2a69a67b46ab.png)
Or are you talking about something else ?

octaeder commented:
Really scary. That's what I see (the last one from your screenshots):
![grafik](https://user-images.githubusercontent.com/102688820/167218734-3e8adf06-3523-43ae-9899-c2cd21ae8389.png)
Did I forget to press a button or so? I can't see any. Do I have to address you directly with @...? I thought that pending means that I have to wait for your answer. Btw I see it should be log panel not Log tab. Maybe I should copy things to separate comments.

sunderme commented:
Looks like the same issue which I had. There was a "submit review" button on the files pages.

octaeder commented:
@sunderme please review

octaeder commented:
Maybe I should change the following in chapter 0:
"On the lower right is a "message panel" where you find tabs for messages, the build log, and so on."
to
"On the lower right you see a "messages panel". You can switch to the "log panel", the "review panel", or the "search results panel" there. "
and add a link for "log panel" to section 3.2 "The log files".

sunderme commented:
Thanks.
I suggest new PRs for other changes. The smaller the PR the better as it allows more precise discussions.
And I assume you "preview" panel above.
